fv3gfs/util/buffer.py

```diff
@@ -2,6 +2,7 @@
 from ._timing import Timer, NullTimer
 from numpy import ndarray
 import contextlib
+import cupy as cp
```

Comment:
Need to contain this in a try: except ModuleNotFoundError block, with the appropriate change to logic below
```suggestion
try:
    import cupy as cp
except ModuleNotFoundError:
    cp = None
```